region/brinstar/green/Brinstar Pre-Map Room.json

```diff
@@ -229,26 +201,79 @@
     },
     {
       "link": [2, 1],
-      "name": "Speedball",
+      "name": "Speedball (Come In Shinecharging)",
+      "notable": false,
+      "entranceCondition": {
+        "comeInShinecharging": {
+          "length": 11,
+          "openEnd": 0
+        }
+      },
+      "requires": [
+        "canSpeedball"
+      ],
+      "clearsObstacles": ["A"],
+      "devNote": "There are 3 unusable tiles in this runway."
+    },
+    {
+      "link": [2, 1],
+      "name": "Speedball (In-Room)",
       "notable": false,
       "requires": [
-        "canSpeedball",
-        {"or": [
-          {"canComeInCharged": {
-            "framesRemaining": 180,
-            "fromNode": 2,
-            "unusableTiles": 3
-          }},
-          {"and": [
-            "canSlowShortCharge",
-            {"canShineCharge": {
-              "usedTiles": 14,
-              "openEnd": 1
-            }}
-          ]}
-        ]}
+        "canSlowShortCharge",
+        {"canShineCharge": {
+          "usedTiles": 14,
+          "openEnd": 1
+        }},
+        "canSpeedball"
       ],
-      "clearsObstacles": ["A"]
+      "clearsObstacles": ["A"],
+      "note": "Run on the upper platform to perform the speedball.",
+      "devNote": "FIXME: This length implies 3 unusable tiles, yet the docs say 5 for a slow speedball in general?"
+    },
+    {
+      "link": [2, 1],
+      "name": "Temporary Blue Bounce",
+      "notable": false,
+      "requires": [
+        {"canShineCharge": {
+          "usedTiles": 17,
+          "openEnd": 1
+        }},
+        "canTemporaryBlue",
+        "canSpringBallBounce"
+      ],
+      "clearsObstacles": ["A"],
+      "note": "Use the upper platform to run and gain temporary blue."
+    },
+    {
+      "link": [2, 2],
+      "name": "Leave With Runway",
+      "notable": false,
+      "requires": [],
+      "exitCondition": {
+        "leaveWithRunway": {
+          "length": 15,
+          "openEnd": 0
+        }
+      }
+    },
+    {
+      "link": [2, 2],
+      "name": "Leave Shinecharged",
```

Comment:
[2,1] is able to leave shinecharged using the same runway.  60-65 frames remaining.
Strange how that would be the only reason to have `A` and there is no strat which uses it.

Reply:
Added a 2->1 leaveShinecharged strat now. I'm guessing the "A" obstacle was mainly for the scenario where you go to the other side to kill the enemies to unlock the gray door, then go back. This can also be useful with ammo doors like in Map Rando (though right now the randomizer is not very smart in how it handles doors -- doesn't remember they're unlocked, but we'll fix that eventually).